src/quacc/utils/files.py

```diff
@@ -23,6 +23,10 @@
     Filenames = str | Path | list[str | Path]
     SourceDirectory = str | Path
 
+    SourceDirectorySchema = (
+        SourceDirectory | dict[str, SourceDirectory] | list[SourceDirectory]
+    )
+
```

Comment:
@tomdemeyere: Throughout your update to this PR, let's not lump the type hints together like this, as it is very confusing. 

A "schema" means something quite specific --- it is a structured output (typically JSON style) with various attributes and corresponding values.

Since this is a type hint that is visible to end-users through the recipes themselves, we want to be as clear as possible. In this case, explicitly enumerating the three options is better in my opinion.

Reply:
You are right it does not make sense, I don't know what I did here... I should probably take more breaks 😅

Reply:
Not me looking at the timezone difference right now... 👀

Reply:
Ah yes! this happens from time to time 💀